src/builderRegistry/utils/shared.ts

```diff
@@ -0,0 +1,137 @@
+  rewardReceiver_: Address,
+  rewardPercentage_: BigInt
+): void {
+  const builder = Builder.load(builder_);
+  if (builder == null) return;
+  builder.rewardReceiver_ = rewardReceiver_;
+  builder.backerRewardPercentage_ = rewardPercentage_;
+  let builderState = BuilderState.load(builder.id);
+  if (builderState == null) {
+    builderState = new BuilderState(builder.id);
+    builderState.builder_ = builder.id;
+    builderState.kycPaused_ = false;
+    builderState.selfPaused_ = false;
+    builderState.communityApproved_ = false;
+  }
+  builderState.kycApproved_ = true;
+  builderState.initialized_ = true;
+
+  builderState.save();
+  builder.save();
+}
+
+export function rewardReceiverUpdated(
+  builder_: Address,
+  newRewardReceiver_: Address
+): void {
+  const builder = Builder.load(builder_);
+  if (builder == null) return;
+  builder.rewardReceiver_ = newRewardReceiver_;
+
+  builder.save();
+}
+
+export function communityApproved(builder_: Address): void {
+  const builder = Builder.load(builder_);
+  if (builder == null) return;
+  let builderState = BuilderState.load(builder.id);
+  if (builderState == null) {
+    builderState = new BuilderState(builder.id);
+    builderState.builder_ = builder.id;
+    builderState.kycApproved_ = false;
+    builderState.kycPaused_ = false;
+    builderState.selfPaused_ = false;
+    builderState.initialized_ = false;
+    builderState.pausedReason_ = "";
+  }
+  builderState.communityApproved_ = true;
+
+  builderState.save();
+  builder.save();
+}
+
+export function communityBanned(builder_: Address): void {
+  const builderState = BuilderState.load(builder_);
+  if (builderState == null) return;
+  builderState.communityApproved_ = false;
+  builderState.save();
+}
+
+export function kycApproved(builder_: Address): void {
+  const builderState = BuilderState.load(builder_);
+  if (builderState == null) return;
+  builderState.kycApproved_ = true;
+  builderState.save();
+}
+
+export function kycRevoked(builder_: Address): void {
+  const builderState = BuilderState.load(builder_);
+  if (builderState == null) return;
+  builderState.kycApproved_ = false;
+
+  builderState.save();
+}
+
+export function kycPaused(builder_: Address): void {
+  const builderState = BuilderState.load(builder_);
+  if (builderState == null) return;
+  builderState.kycPaused_ = true;
+
+  builderState.save();
+}
+
+export function kycResumed(builder_: Address): void {
+  const builderState = BuilderState.load(builder_);
+  if (builderState == null) return;
+  builderState.kycPaused_ = false;
+  builderState.save();
+}
+
+export function selfPaused(builder_: Address): void {
+  const builderState = BuilderState.load(builder_);
+  if (builderState == null) return;
+  builderState.selfPaused_ = true;
+
+  builderState.save();
+}
+
+export function selfResumed(builder_: Address): void {
+  const builderState = BuilderState.load(builder_);
+  if (builderState == null) return;
+  builderState.selfPaused_ = false;
```

Comment:
Shouldn't we update also the backer reward percentage here?

Reply:
I need to validate this since I was thinking in something different.

Reply:
updated